src/machine/board_hifive1b.go

```diff
@@ -40,3 +42,21 @@ const (
 	UART_TX_PIN = NoPin
 	UART_RX_PIN = NoPin
 )
+
+// SPI pins
+const (
+	SPI0_SCK_PIN  = NoPin
+	SPI0_MOSI_PIN = NoPin
+	SPI0_MISO_PIN = NoPin
+
+	SPI1_SCK_PIN  = D13
+	SPI1_MOSI_PIN = D11
+	SPI1_MISO_PIN = D12
+)
+
+// SPI on the HiFive1.
+var (
+	SPI1 = SPI{
+		Bus: sifive.QSPI1,
```

Comment:
This will break simulation on https://play.tinygo.org because it can't compile with the WebAssembly backend. I think it's best to move the `SPI1` variable to machine_fe310.go, or alternatively to a file that is only included in the baremetal build (example: board_hifive1b_baremetal.go).

(Sidenote: a similar change will be needed for samd-based boards to allow them to run on https://play.tinygo.org).

Reply:
Might be better to change the playground since there are other boards that SPI0 cannot be used due to various unchangeable defaults on that hardware.

We could also add a `SPIDefault` that points to the default SPI to be used  on any given board. This would require changing all the boards to add, but would certainly be a lot more obvious what is going on.

Reply:
> Might be better to change the playground since there are other boards that SPI0 cannot be used due to various unchangeable defaults on that hardware.

How exactly would that look?

At the moment, it works as follows:

* board files (like board_pca10040.go) specify just pin mappings and nothing else. Can compile on wasm.
* chip files (like machine_nrf52.go) specify hardware-dependent details. For example, for the nrf series, it specifies `SPI0` and `SPI1`. Is not able to compile for wasm, but machine_generic.go provides a solution there for wasm.

I realize that this conflates two somewhat distinct things: the board/chip distinction, and the chip-specific/simulation distinction. This happens to work for the nrf family, but is more problematic for other chip families.

One solution is to split board definitions into two files (so you have 3 files for every chip/board combo):

* board_hifive1b.go: contains pin mappings that compile natively and on wasm
* board_hifive1b_baremetal.go: contains the parts that require the sifive package like SPI1.
* board_generic.go: provides a generic SPI0, SPI1, etc.
* machine_fe310.go: provides the `SPI` type, etc.

Reply:
OK I think your original suggestion of moving the definition to `machine_fe310.go` is probably best. First of all, it is simplest. But also, it is correct, in that the same SPI register mapping exist on all of the fe310-based boards.

The solution with 3 files per board seems the most complex, probably to be avoided if possible for our friendly implementer's sake.